### ISSUE TITLE
Envoy DNS resolution appears to reset circuit breaking

### ISSUE THREAD

**Issue by UnwashedMeme** [istio/istio#8673]
**Describe the bug**
Outlier detection removes a member from the cluster, but it gets returned every 4 or 5 seconds effectively negating outlier detection.

**Expected behavior**
The ejected member to remain excluded for at least `baseEjectionTime` (or a multiple of it).

**Steps to reproduce the bug**

1. Create a `ServiceEntry` for an external service
  * `resolution: DNS`
  * multiple endponts  -- I have 3, one of which *always* returns a 503.
1. Create a `DestinationRule` covering this service with a `trafficPolicy.outlierDetection`  block.
1. VirtualService, etc.
1. Request that virtual service a lot: `while true; do curl $SVC`
  * It continues to return 503s at slightly less than 1/3 the time.

Watching the proxy's `/stats` endpoint I observe (on the specific `cluster.outbound|....`)
* `.lb_healthy_panic: 0` -- it isn't keeping the endpoint in because it is panic mode
* `.outlier_detection.ejections_detected_consecutive_gateway_failure` does increase
* `.ejections_active` never stays non-zero for greater than a few seconds
* `.membership_change` increases by 3 regularly, about every 4 to 5 seconds.  Every time that period comes around the `.ejections_active` resets. This number gets very big over time, e.g 12k after a couple hours.

Reading https://groups.google.com/forum/#!topic/istio-dev/26Gn8oHYHEE it looks like istio/envoy may have changed from pulling config to pushing config relatively recently.  It looks like every 4 or 5 seconds istio pushes new config to envoy, which resets its data about that cluster member being an outlier.

**Version**

* istio download 1.0.1, updated `global.tag: 1.0.2` in the values.yaml given to helm
* kubernetes:  1.11.2

**Installation**

helm

**Environment**

azure, created with acs-engine.

**Comment by UnwashedMeme** [comment]
For anyone else that finds this, my work around was to change resolution from DNS to static and put the address in myself, with this the `membership_change` stat stabilized and `ejections_active` remains at 1. My 503s have gone down because it doesn't keep trying to re-add the host that is erring.

**Comment by UnwashedMeme** [comment]
so based on the above DNS -> STATIC find I found a few more items.

envoy `/config_dump` shows me that it has the individual hostnames I entered in the ServiceEntry as the hosts in the cluster. So I don't think this is an istio issue about pushing config, I think this is envoy handling dns resolution.

When I look in `/clusters` i see the backend ip addresses but the stats about how many requests it has handled get reset frequently (ever 4 or 5 seconds) and as such the outlier detection doesn't work.

Each hostname is a single A record that doesn't change, so based on the description at https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/service_discovery#strict-dns  it shouldn't be draining the connection pool, yet here we are.
